packages/browser/src/core/user/index.ts

```diff
@@ -398,3 +398,53 @@ export class Group extends User {
     return undefined
   }
 }
+
+export class UniversalStorage {
+  private cookies: Store
+  private localStorage: Store
+  private mem: Store
+
+  options: UserOptions = {}
+
+  constructor(isDisabled: boolean, cookieOptions?: CookieOptions) {
+    this.localStorage =
+      isDisabled || !LocalStorage.available()
+        ? new NullStorage()
+        : new LocalStorage()
+
+    this.cookies =
+      !isDisabled && Cookie.available()
+        ? new Cookie(cookieOptions)
+        : new NullStorage()
+
+    this.mem = isDisabled ? new NullStorage() : new Store()
+
+    autoBind(this)
+  }
+
+  chainGet<T>(key: string): T | null {
```

Comment:
The word "chain" and "try" seem like implementation details, any reason to not continue to use the conventions of the web storage API (if not outright implementing it?).